diff --git startup.sh startup.sh
index 9bb4a87..8031977 100755
--- startup.sh
+++ startup.sh
@@ -25,9 +25,11 @@ echo '[docker setup]'
 distro=$(cat /etc/*release | egrep '^ID=' | awk -F= '{print $2}' | tr -d \")
 echo $distro detected.
 if [ x"$distro" == x"centos" ]; then
-   yum install -y docker
+   yum install -y yum-utils device-mapper-persistent-data lvm2
+   yum-config-manager --add-repo https://download.docker.com/linux/centos/docker-ce.repo
+   yum install -y docker-ce
+   echo 'OPTIONS="--storage-opt dm.basesize=20G"' >> /etc/sysconfig/docker
    systemctl stop firewalld || true
-   sed -i 's/DOCKER_STORAGE_OPTIONS=/DOCKER_STORAGE_OPTIONS=--storage-opt dm.basesize=200G /g' /etc/sysconfig/docker-storage
    systemctl start docker
 fi
 
